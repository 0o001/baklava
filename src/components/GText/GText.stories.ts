import { storiesOf } from '@storybook/vue';
import GText from './GText.vue';

storiesOf('GText', module)
  .addParameters({ component: GText })
  .add('Headline-01', () => ({
    components: { GText },
    template: `
      <GText variant="headline-01" color="orange-500"> Rubik Bold, 64px</GText>`,
  }))
  .add('Headline-02', () => ({
    components: { GText },
    template: `
      <GText variant="headline-02"> Rubik Light, 48px</GText>`,
  }))
  .add('Headline-03', () => ({
    components: { GText },
    template: `
      <GText variant="headline-03"> Rubik Light, 30px</GText>`,
  }))
  .add('Headline-04', () => ({
    components: { GText },
    template: `
      <GText variant="headline-04"> Rubik Medium, 24px</GText>`,
  }))
  .add('subtitle-01', () => ({
    components: { GText },
    template: `
      <GText variant="subtitle-01"> Rubik Medium, 20px</GText>`,
  }))
  .add('subtitle-02', () => ({
    components: { GText },
    template: `
      <GText variant="subtitle-02"> Rubik Medium, 16px</GText>`,
  }))
  .add('subtitle-03', () => ({
    components: { GText },
    template: `
      <GText variant="subtitle-03"> Rubik Medium, 14px</GText>`,
  }))
  .add('body', () => ({
    components: { GText },
    template: `
      <GText variant="body"> Rubik Regular, 14px</GText>`,
  }))
  .add('caption', () => ({
    components: { GText },
    template: `
      <div>
        <GText variant="caption"> Rubik Regular, 12px</GText>
        <GText variant="caption" inline> Rubik Regular, 12px Inline</GText>
      </div>
    `,
  }))
  .add('menu-item', () => ({
    components: { GText },
    template: `
      <GText variant="menu-item"> Rubik Medium, 12px</GText>`,
  }))
  .add('small', () => ({
    components: { GText },
    template: `
      <GText variant="small"> Rubik Regular, 10px</GText>`,
  }))
  .add('with text align', () => ({
    components: { GText },
    template: `
    <div :style="{ width: '300px' }">
      <GText align="right" variant="body"> Rubik Regular, 14px</GText>
      <GText align="center" variant="body"> Rubik Regular, 14px</GText>
      <GText align="left" variant="body"> Rubik Regular, 14px</GText>
    </div>`,
  }));
