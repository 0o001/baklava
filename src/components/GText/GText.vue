<template>
  <div :class="classNames">
    <slot />
  </div>
</template>

<script lang="ts">

import $_ from '../../utils/dashUtil.ts';

export default {
  name: 'GText',
  props: {
    variant: {
      type: String,
      default: 'body',
    },
    color: {
      type: String,
      default: 'main-grey-500',
    },
    inline: {
      type: Boolean,
      default: false,
    },
    align: {
      type: String,
      default: null,
    },
  },
  computed: {
    classNames () {
      return {
        [`g-text-${this.color}`]: true,
        [`g-text-align-${this.align}`]: this.align != null,
        [$_(this.variant)]: true,
        '-inline': this.inline,
        'g-text': true,
      };
    },
  },
};
</script>

<style lang="scss" scoped>
.g-text {
  font-family: var(--font-family-base);
  line-height: normal;

  &.-headline-01 {
    font-size: var(--font-size-64);
    font-weight: bold;
    letter-spacing: -2px;
  }

  &.-headline-02 {
    font-size: var(--font-size-48);
    font-weight: normal;
    letter-spacing: normal;
  }

  &.-headline-03 {
    font-size: var(--font-size-30);
    font-weight: 300;
    letter-spacing: normal;
  }

  &.-headline-04 {
    font-size: var(--font-size-24);
    font-weight: 500;
    letter-spacing: normal;
  }

  &.-subtitle-01 {
    font-size: var(--font-size-20);
    font-weight: 500;
    letter-spacing: normal;
  }

  &.-subtitle-02 {
    font-size: var(--font-size-16);
    font-weight: 500;
    letter-spacing: normal;
  }

  &.-subtitle-03 {
    font-size: var(--font-size-14);
    font-weight: 500;
    letter-spacing: normal;
  }

  &.-body {
    font-size: var(--font-size-14);
    letter-spacing: normal;
  }

  &.-caption {
    font-size: var(--font-size-12);
    letter-spacing: normal;
  }

  &.-menu-item {
    font-size: var(--font-size-12);
    font-weight: 500;
    letter-spacing: normal;
  }

  &.-small {
    font-size: var(--font-size-10);
    letter-spacing: normal;
  }

  &.-inline {
    display: inline;
  }
}
</style>
