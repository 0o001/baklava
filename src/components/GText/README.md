# GText

## Slots

| Name      | Description |
| --------- | ----------- |
| `default` | &nbsp;      |

## Props

| Name      | Type      | Description | Default           |
| --------- | --------- | ----------- | ----------------- |
| `variant` | `String`  |             | `"body"`          |
| `color`   | `String`  |             | `"main-grey-500"` |
| `inline`  | `Boolean` |             | `false`           |
| `align`   | `String`  |             | `null`            |

## Computed Properties

| Name         | Type     | Description                                             |
| ------------ | -------- | ------------------------------------------------------- |
| `classNames` | `object` | **Dependencies:** `color`, `align`, `variant`, `inline` |

