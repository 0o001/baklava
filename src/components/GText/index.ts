import GText from './GText.vue';

export default GText;
