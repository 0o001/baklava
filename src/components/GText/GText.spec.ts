import { shallowMount, Wrapper } from '@vue/test-utils';
import GText from './';

describe('GText', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallowMount(GText);
  });

  it('should match snapshot when show is prop is given as false ', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render base color as default when color prop does not set', () => {
    expect(wrapper.vm.classNames).toEqual({ '-body': true, 'g-text': true, 'g-text-align-null': false, 'g-text-main-grey-500': true, '-inline': false });
  });

  it('should render given color when color prop set', async () => {
    // given
    wrapper.setProps({
      color: 'green-500',
    });

    // when
    await wrapper.vm.$nextTick();

    // then
    expect(wrapper.vm.classNames).toEqual({ '-body': true, 'g-text': true, 'g-text-align-null': false, 'g-text-green-500': true, '-inline': false });
  });

  it('should render given text-align when align prop set', async () => {
    // given
    wrapper.setProps({
      align: 'center',
    });

    // when
    await wrapper.vm.$nextTick();

    // then
    expect(wrapper.vm.classNames).toEqual({
      '-body': true,
      'g-text': true,
      'g-text-main-grey-500': true,
      'g-text-align-center': true,
      '-inline': false,
    });
  });
});
