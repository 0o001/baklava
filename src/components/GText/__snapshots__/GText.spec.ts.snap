// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GText should match snapshot when show is prop is given as false  1`] = `
<div
  class="g-text-main-grey-500 -body g-text"
/>
`;
